test/error.test.ts

```diff
@@ -0,0 +1,61 @@
+import { it, expect } from 'vitest';
+import { ErrorHTTP, fastErrorHTTP } from '../src';
+
+it('ErrorHTTP', () => {
+  const defaultErrorHTTP = new ErrorHTTP();
+  expect(defaultErrorHTTP.status).toBe(500);
+  expect(defaultErrorHTTP.message).toBe('Internal Server Error');
+  expect(defaultErrorHTTP.stack).toContain('Error: Internal Server Error');
+
+  const statusAsMessage = new ErrorHTTP(404);
+  expect(statusAsMessage.status).toBe(404);
+  expect(statusAsMessage.message).toBe('Not Found');
+  expect(statusAsMessage.stack).toContain('Error: Not Found');
+
+  const messageAndStatus = new ErrorHTTP('Wrongo', 401);
+  expect(messageAndStatus.status).toBe(401);
+  expect(messageAndStatus.message).toBe('Wrongo');
+  expect(messageAndStatus.stack).toContain('Error: Wrongo');
+});
+
+it('FastErrorHTTP', () => {
+  expect(() => {
+    (fastErrorHTTP as any)();
+  }).toThrowError(/code is required to be a string or number/);
+  expect(() => {
+    (fastErrorHTTP as any)(function() {});
+  }).toThrowError(/code is required to be a string or number/);
+
+  const CustomErrorType = fastErrorHTTP('MyCustomErrorTypeCode', 401);
+  const defaultErrorHTTP = new CustomErrorType();
+  expect(defaultErrorHTTP.code).toBe('MyCustomErrorTypeCode');
+  expect(defaultErrorHTTP.status).toBe(401);
+  expect(defaultErrorHTTP.message).toBe('Unauthorized');
+  expect(defaultErrorHTTP.stack).toContain('Error: Unauthorized');
+
+  // @ts-ignore
+  const statusAsMessage = new CustomErrorType(404);
+  expect(statusAsMessage.code).toBe('MyCustomErrorTypeCode');
+  expect(statusAsMessage.status).toBe(401);
+  expect(statusAsMessage.message).toBe('404');
```

Comment:
Slightly confused by the usage here, is the `404` that you're passing into `CustomErrorType` the message and not the status code? Does it convert the number to a string?

Reply:
Yeah this one was a little confusing to me as well, but the premise here is that the custom error type should always use the original and intended status code (401 in this case). By passing in a numeric value to the custom-made class it gets [passed to this `format` line](https://github.com/mapbox/mapbox-error/blob/1514642bd0136f9de37aba88c270aba63487ba1a/src/error.ts#L48), which does the string conversion for us, and doesn't override the HTTP status.